tdesign/src/Layout/Header/HeaderLeft.tsx

```diff
@@ -1,5 +1,5 @@
 import * as React from "react";
-import { Box } from "@material-ui/core";
+import { Box } from "@mui/material";
```

Comment:
General comment:

It's disappointing to see `Box` used as a `div` with a `className`. Importing MUI only to use it like a `div`, sounds like an overkill and will make it harder to migrate away from MUI. Maybe we're overusing `Box` across the app :smile: ?

Reply:
The point of this branch is to upgrade MUI, not change code styling.

Reply:
Hence the "general comment" in the first line of the comment. I'm picking out patterns from the code that are smelly to me to think if we should do things differently in the future and to gather opinions. Resolving the thread kills the opinion-gathering part. I can be on the lookout for such uses in the future and comment on it then to spark the discussion later, but it seems artificial to me

Reply:
The main reason for using `Box` over `div` is because `div` does not support the `sx` prop. Now that we're using `styled-engine`, we can prefer that over `Box`.

Reply:
Not also that, but we can also use `div` instead of `Box` if we just need it for a container. For example in this file (`HeaderLeft`) we might as well use a div because there is nothing `Box`-specific we need:

```tsx
const HeaderLeft = ({ children }: IHeaderLeftProps) => {
  return <div className="header--left">{children}</div>;
};
```

Just saying to consider doing that in the future to avoid overusing MUI :smile:

Reply:
Right, in cases where a container definitely doesn't need to access the theme (like `HeaderLeft`) `div` seems fine.

Of course when we implement designer comps, its pretty common to run into situations where we need access to the theme e.g. maybe there's something light/dark mode reactive. IME that's when the need for `sx` or `styled()` enters the picture pretty reliably